lib/ranking.rb

```diff
@@ -0,0 +1,53 @@
+test = { 
+  url: {
+      "www.bbc.co.uk"=> 1, "www.bbc.co.uk/sports"=> 2
+    },
+    keywords: {
+      "news"=> 2 , "sports"=> 5
+    },
+    description: {
+      "bbc"=> 5 , "football"=> 7
+    },
+
+    headers: {
+      "fire"=>1, "goal"=>4 
+    },
+    text: {
+      "referee"=>2, "post"=> 5, "water"=>4, "eyes"=>3, "sports"=> 1
+    }
+  }
+
+class Ranker
+attr_reader :word_score, :hash
+
+  def initialize(hash, word)
+    @hash = hash
+    @word_score = 0
+    @word = word
+  end
+
+
+  def calculate_total_score
+    @hash.each do|k,v| 
```

Comment:
I think is better if the placeholders have more meaningful names.

Ex:
key, value